asllib/doc/ASLSemanticsReference.tex

```diff
@@ -19,160 +283,1634 @@
+The semantic relations are constructively defined via \emph{semantic rules}.
+The rules, in turn, inductively define transitions from configurations with an AST node
+in terms of transitions starting from configurations with its child nodes.
+
+\paragraph{Execution.}
+A valid execution of an ASL specification transitions from an \emph{initial configuration},
+which consists of the given specification and the standard library specification, to an output configuration
+consisting of an output value and a \herd\ execution graph.
+
+\paragraph{Primitive Subprograms.}
+The semantics of ASL is parameterized by a set of primitive subprograms ---
+subprograms whose implementation is not defined by an ASL body and whose effect on the dynamic environment
+is defined externally. Critically, access to memory is given by primitive subprograms.
+
+We define two types of semantics --- a \emph{sequential semantics} and a \emph{concurrent semantics}.
+
+\paragraph{Concurrent Semantics.}
+The concurrent semantics includes \herd\ execution graphs.
+Intuitively, these graphs define read and write effects to variables and constraints over those effects.
+Together with the constraints that define the ARM memory model~\cite{AlglaveDGHM21}, these graphs axiomatically define
+the valid sequences of read and write operations induced by a given specification.
+
+\paragraph{Sequential Semantics.}
+The sequential semantics corresponds to executing an ASL specification in the context of a single (conceptual) process
+of execution --- notice that ASL does not contain any concurrency constructs. Access to memory is hidden away
+in primitive subprograms that interface with a map-like memory implementation that does not reorder memory accesses
+and serves one access at a time.
+%
+Technically, the sequential semantics is defined by dropping the \herd\ execution graph components
+from the configurations.
+
+\section{Mathematical Notations and Conventions}
+
+\input{ASLNotations.tex}
+
+\paragraph{OCaml-style Notations.}
+We use the notation $L(v_{1..k})$, where $L$ is a label and $v_{1..k}$ is a (possibly-singleton) tuple of abstract values,
+to denote the tuple $(L,v_{1..k})$.
+We also write $L(T_{1..k})$, where $T_{1..k}$ denote mathematical type of values, to stand for
+$\{L\} \times T_1 \times \ldots \times T_k$.
+%
+We later use these notations to define \emph{semantic configurations} and \emph{native values}, \underline{not to be confused with labelled AST nodes}.
+%
+The notation $\langle \cdot \rangle$ stands for empty-or-singleton sets, where $\None\triangleq\langle\rangle$ denotes
+an empty set and $\langle v \rangle$ denotes a set containing the single element $v$.
+%
+The notation $\langle T \rangle$, where $T$ denotes a mathematical type of values, stands for
+$\{ \langle\rangle \} \cup \{\langle v \rangle \;|\; v \in T\}$.
+
+\paragraph{Misclleaneous.}
```

Comment:
typo: missing "e"
also throughout, every now and then you've put a full stop at the end of the section title - please remove them

Reply:
I used period to end titles of paragraphs, since they are inline with the content of the paragraph and I like separating those. It's a matter of style.

Reply:
ok, then please make sure you have consistency throughout - that didn't seem to be the case on the version I reviewed